include/SDL3/SDL_events.h

```diff
@@ -615,7 +620,21 @@ typedef struct SDL_GamepadButtonEvent
 /**
  * Gamepad device event structure (event.gdevice.*)
  *
+ * Joysticks that are supported Gamepads receive both an SDL_JoyDeviceEvent and
+ * an SDL_GamepadDeviceEvent.
+ *
+ * SDL will populate the event queue with GAMEPAD_ADDED events for joysticks
+ * that are already plugged in duing SDL_Init and are recognized as Gamepads.
+ *
+ * Additionally, when new gamepad mappings are loaded via
+ * SDL_AddGamepadMappingsFromFile, SDL will generate GAMEPAD_ADDED events for
+ * these devices as well.
+ *
  * \since This struct is available since SDL 3.0.0.
+ *
+ * \sa SDL_AddGamepadMappingsFromFile
```

Comment:
If you're going to add functions that can generate these events, there are several, as well as another hint.

Reply:
It's probably better to document those functions to note that they'll generate the events instead.